text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,284 @@
+achieved:
+
+### Resource detectors (soon to be entity detectors) need to be composable / disjoint
+
+Entity detection and Resource Manager now fulfill this need.
+
+### New entities added by extension should not break existing code
+
+Users will need to configure a new Entity detector for new entities being modelled.
+
+### Navigational attributes need to exist and can be used to identify an entity but could be augmented with UUID or other aspects. - Having ONLY a UUID for entity identification is not good enough
+
+Resource will still be composed of identifying and descriptive attributes of Entity, allowing baseline navigational attributes users already expect from resource.
+
+### Collector augmentation / enrichment (resource, e.g.) - Should be extensible and not hard-coded. We need a general algorithm not specific rulesets
+
+Entity concept provides a new "bundle" mechanism to resource for the Collector to augment enrich a group of attributes and better identify conflicts (or identity changes) caused therein.
+
+### Users are expected to provide / prioritize "detectors" and determine which entity is "producing" or most-important for a signal
+
+The Resource Manager allows users to configure priority of Entity Detectors.
+
+### For an SDK - ALL telemetry should be associated with the same set of entities (resource labels)
+
+Resource Manager is responsible for resolving entities into a cohesive Resource that meets the same demands as Resource today.
+
+## Open Questions
+
+The following remain open questions:
+
+### How to attach Entity "bundle" information in Resource?
+
+The protocol today requires a raw grab bag of Attributes on Resource. We cannot break this going forward.  However, Entities represent a new mechanism of "bundling" attributes on Resource and interacting with these bundles.  We do not want this to bloat the protocol, nor do we want it to cause oddities.
+
+Going forward, we have set of options:
+
+- Duplicate attributes in `Entity` section of Resource.
+- Reference attributes of Resource in entity.
+- Only identify Entity id and keep attribute<->entity association out of band.
+- Extend Attribute on Resource so that we can track the entity type per Key-Value (across any attribute in OTLP).
+
+The third option prevents generic code from interacting with Resource and Entity without understanding the model of each.  The first keeps all usage of entity simple at the expense of duplicating information and the middle is awkward to interact with from an OTLP usage perspective. The fourth is violates our stability policy for OTLP.
+
+### How to deal with Resource/Entities whose lifecycle does not match the SDK?
+
+This proposal motivates a Resource Coordinator in the SDK whose job could include managing changes in entity lifetimes, but does not account for how these changes would be broadcast across TracerProvider, LogProvider, MeterProvider, etc.  That would be addressed in a follow on OTEP.
```

Comment:
I think we don't need to broadcast anything in the TracerProvider, LogProvider, MeterProvider if we exclude descriptive attributes from the EntityRef. By definition the Type and Id of entity can't change during entity's lifetime so the EntityRef is immutable, there is no changes to broadcast.

One case that we may want to broadcast is if an entity that was previously included in a Resource via EntityRef ceases to exist. Arguably in this case we want to stop including it in the Resource. I don't know though if there is a real use case like this where the SDK outlives the Entity that was detected by SDK detector. Do we have such cases?

Reply:
We need to broadcast when an Entity no longer exists, at a minimum, or when a new one has replaced the previous value (e.g. Browser Session).

The main focus here is dealing with entities whose lifetimes do NOT match the SDK.

Regarding mutable/descirptive attributes let's talk more in the entities WG.  There was adamant discussion that these are needed in Resource.

Reply:
>We need to broadcast when an Entity no longer exists, at a minimum, or when a new one has replaced the previous value (e.g. Browser Session).

Are we thinking about Browser Session as an entity? It does not seem to be an entity to me. Instead it seems more like an execution context and the way you would want to record this information is via Span/Metric/LogRecord attributes, not via Resource or Entity attributes. Isn't that what the [session semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/session.md) say (that it is a LogRecord attribute)?

Reply:
>Regarding mutable/descirptive attributes let's talk more in the entities WG. There was adamant discussion that these are needed in Resource.

To be clear: I think it is fine to include descriptive attributes in the Resource `attributes` field _as long as_ you can ensure these attributes don't mutate during Resource lifetime. If the attributes mutate then we have a problem since Resource is assumed immutable in the SDK spec and in all current SDK implementations. I don't know how we would overcome this problem without breaking the Resource API (perhaps there is a way?). 

The reason I think it is right to include descriptive attributes in the Resource is because that's what we do today and I think we must continue doing so for backward compatibility reasons so that recipients continue seeing the same data in the Resource.

That said, I don't see why we would _also_ want to include those same descriptive attributes in the EntityRef. I don't see the purpose. If there is a purpose I would like to know.